src/test/java/uk/nhs/hee/tis/trainee/sync/CachingIntTest.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright 2021 Crown Copyright (Health Education England)
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and
+ * associated documentation files (the "Software"), to deal in the Software without restriction,
+ * including without limitation the rights to use, copy, modify, merge, publish, distribute,
+ * sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT
+ * NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+ * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+package uk.nhs.hee.tis.trainee.sync;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.amazonaws.services.sqs.AmazonSQS;
+import java.util.Optional;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.context.TestConfiguration;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.cache.Cache;
+import org.springframework.cache.CacheManager;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Primary;
+import uk.nhs.hee.tis.trainee.sync.config.MongoConfiguration;
+import uk.nhs.hee.tis.trainee.sync.model.Operation;
+import uk.nhs.hee.tis.trainee.sync.model.Post;
+import uk.nhs.hee.tis.trainee.sync.repository.PostRepository;
+import uk.nhs.hee.tis.trainee.sync.service.PostSyncService;
+
+
+@SpringBootTest
+class CachingIntTest {
+
+  private static final String POST_FORDY = "fordy";
+
+  // We require access to the mock before the proxy wraps it.
+  private static PostRepository mPostRepository;
+
+  @Autowired
+  PostSyncService postSyncService;
+
+  @Autowired
+  CacheManager cacheManager;
+
+  private Cache postCache;
+
+  private Post post;
+
+  @BeforeEach
+  void setup() {
+    post = new Post();
+    post.setTisId(POST_FORDY);
+    post.setOperation(Operation.DELETE);
+    post.setTable(Post.ENTITY_NAME);
+
+    postCache = cacheManager.getCache(Post.ENTITY_NAME);
```

Comment:
I think it needs `postCache.clear()` so that the cache starts blank at every test, because the cache might have persisted some objects from running tests before.